include/nbl/builtin/hlsl/complex.hlsl

```diff
@@ -379,6 +379,19 @@ complex_t<Scalar> rotateRight(NBL_CONST_REF_ARG(complex_t<Scalar>) value)
     return retVal;
 }
 
+// Annoyed at having to write a lot of boilerplate to do a select
+// Essentially returns what you'd expect from doing `condition ? a : b`
+template<typename Scalar>
+complex_t<Scalar> ternaryOperator(bool condition, NBL_CONST_REF_ARG(complex_t<Scalar>) a, NBL_CONST_REF_ARG(complex_t<Scalar>) b) 
```

Comment:
make a functor similar to `plus_assign`, `plus` etc. thats called `ternary_operator`

I think following might work for `!is_fundamental_v<T>` types
```cpp
if (condition)
   return a;
else
   return b;
```

Reply:
Checked godbolt, using `?` with vector types compiles down to a single select, doing `if (condition) else` compiles down to two selects :/

Reply:
btw soon `lerp`/`mix` with a boolean mixer can be used to work around the ternary shit on structs